specs/order_spec.rb

```diff
@@ -141,21 +183,90 @@
     end
 
     it "Returns accurate information about the last order" do
-      # TODO: Your test code here!
+      id = 100
+      products = {
+        "Amaranth" => 83.81,
+        "Smoked Trout" => 70.6,
+        "Cheddar" => 5.63
+      }
+      customer_id = 20
+      fulfillment_status = :pending
+
+      order = Order.all.last
+
+      # Check that all data was loaded as expected
+      expect(order.id).must_equal id
+      expect(order.products).must_equal products
+      expect(order.customer).must_be_kind_of Customer
+      expect(order.customer.id).must_equal customer_id
+      expect(order.fulfillment_status).must_equal fulfillment_status
     end
   end
 
+
+
   describe "Order.find" do
     it "Can find the first order from the CSV" do
-      # TODO: Your test code here!
+      first = Order.find(1)
+
+      expect(first).must_be_kind_of Order
+      expect(first.id).must_equal 1
     end
 
     it "Can find the last order from the CSV" do
-      # TODO: Your test code here!
+      last = Order.find(100)
+      p last
+      expect(last).must_be_kind_of Order
+      expect(last.id).must_equal 100
     end
 
     it "Returns nil for an order that doesn't exist" do
-      # TODO: Your test code here!
+      expect(Order.find(123456)).must_be_nil
+    end
+
+    it 'Returns array of found orders by customer_id' do
+      expect(Order.find_by_customer(26)).must_be_kind_of Array
+    end
+
+    it 'Returns all order instances that match customer_id' do
+      # puts Order.find_by_customer(26)
+      expect((Order.find_by_customer(26)).size).must_equal 6
+    end
+
+    it 'Returns correct order instance that match customer_id' do
+      id = 3
+      products = {
+        "Vegetable spaghetti" => 37.83,
+        "Dates" => 90.88,
+        "WhiteFlour" => 3.24,
+        "Caraway Seed" => 54.29
+      }
+      customer_id = 5
+      fulfillment_status = :processing
+      # ensure that only one order is getting returned
+      order_found = Order.find_by_customer(5).first
+      # ap order_found
+      # Check that customer 5 data was loaded into 'found' array as expected
+      expect(order_found).must_be_kind_of Order
+      expect(order_found.id).must_equal id
+      expect(order_found.products).must_equal products
+      expect(order_found.customer).must_be_kind_of Customer
+      expect(order_found.customer.id).must_equal customer_id
+      expect(order_found.fulfillment_status).must_equal fulfillment_status
+    end
+
+    it 'Returns Argument Error if customer id not in order data' do
+      expect{ Order.find_by_customer(666) }.must_raise ArgumentError
+    end
+
+    it 'prints a list of orders' do
```

Comment:
It doesn't need to **print** a list of orders, simply retrieve them.  

So you could do:
```ruby
  orders = Order.find_by_customer(1)
  orders.each do |order|
    expect(order).must_be_instance_of Order
    expect(order.customer).must_equal 1
  end
```